src/efi/efisupport.c

```diff
@@ -110,11 +119,75 @@ static int find_free_boot_rec(void) {
     return res;
 }
 
-int efi_create_boot_rec(void) {
+typedef struct part_info {
+    char *disk_path;
+    int part_no;
+    char *part_type;
+} part_info_t;
+
+/* Given the location of the booloader, returns the partition information needed
+ * to create boot variable which points to that bootloader. */
+int get_part_info(const char *path, part_info_t *pi) {
```

Comment:
I took a look at it. [`probe.h`](https://github.com/busykai/clr-boot-manager/blob/master/src/lib/probe.h) is a high-level API quite specific to kernel installation. there's also [`blkid_stub.h`](https://github.com/busykai/clr-boot-manager/blob/master/src/lib/blkid_stub.h) which is just an indirection to `libblkid` (for testing, I assume). I might end up reusing the latter for the testing sake. In some cases (e.g. `devnode_to_devpath`), vanilla `libblkid` impl of the same is more comprehensive.

P.S. I think you commented on renamed files and the comments appear as outdated. :)